### ISSUE TITLE
Error when trying to access 'variables' inside the PDF template in Craft Commerce

### ISSUE THREAD

**Issue by Stalex89** [craftcms/commerce#2039]
**Description**
I'm trying to include the custom image field inside the PDF file of Craft Commerce. I added the following code to the module:

```
use craft\commerce\events\PdfEvent;
use craft\commerce\services\Pdfs;

  Event::on(
            Pdfs::class,
            Pdfs::EVENT_BEFORE_RENDER_PDF,
            function(PdfEvent $event) {
                // Modify `$event->order`, `$event->option`, `$event->template`,
                // and `$event->variables` to customize what gets rendered into a PDF;
                // or render your own PDF and set its string on `$event->pdf`
                // ...

                $event->variables['productImages'] = array_map(function ($value) {

                    $variant = Plugin::getInstance()->variants->getVariantById($value['purchasableId']);
                    
                    $variantImage = $variant->getFieldValue('productImage')->one();
                    
                    return $variantImage ? $variantImage->getUrl() : null;
                }, $event->order->lineItems);
            }
        );
```
Dumping the variable in the plugin actually gives me the result, so it should be accessible:

![ViRTX](https://user-images.githubusercontent.com/23526237/110293470-e83a2200-7fee-11eb-903b-360bb1f04fe4.png)

But when I'm trying to access the field 'variables' inside the template, it downloads the pdf file with an error text inside the file "An error occurred while generating this PDF" . Meanwhile I can access 'order' variables, and it works fine. Also the prd template works fine without errors when not accessing 'variables'.

I call ```use craft\commerce\services\Pdfs;``` inside the code because ```use craft\commerce\services\Pdf;``` doesn't exist, however it is ```use craft\commerce\services\Pdf;``` that mentioned in the documentation:

![HYJAm](https://user-images.githubusercontent.com/23526237/110294096-b4133100-7fef-11eb-916b-9184a6404185.png)

![Screenshot 2021-03-08 at 09 13 26](https://user-images.githubusercontent.com/23526237/110293913-72828600-7fef-11eb-95de-aa04d5894124.png)

Can you tell me please what can be the cause of the issue here, or I'm doing something wrong? Thank you

![jhOsY](https://user-images.githubusercontent.com/23526237/110293742-3bac7000-7fef-11eb-8a3e-8e5b72adf62a.png)

**To Reproduce**
How, and what happened?
1. Add the code from the above to custom module/plugin which adds some $event->variables  before rendering PDF 
2. Try to access variables inside the pdf template

**Expected behavior**
Variables are accessed

**Additional info**
- Craft version: Craft Pro 3.5.17.1
- Commerce version: 3.2.13.2
- PHP version: 7.4.15
- Database driver & version: MySQL 5.5.68
- Other Plugins & versions: MultiSafepay for Craft Commerce 2.1.0.1

**Additional context**

**Comment by lukeholder** [comment]
I have just made a PR to fix the docs: https://github.com/craftcms/docs/pull/166

It should be `Pdfs` service in the docs.

```
use craft\commerce\services\Pdfs;

  Event::on(
            Pdfs::class,
            Pdfs::EVENT_BEFORE_RENDER_PDF,
```

I tried to write your code where it would work.


```php
Event::on(
    Pdfs::class,
    Pdfs::EVENT_BEFORE_RENDER_PDF,
    function(PdfEvent $event) {

        $variables = $event->variables;
        $productImagesByLineItemId = [];
        foreach ($event->order->getLineItems() as $lineItem)
        {
            /** @var Variant $variant */
            if($variant = $lineItem->getPurchasable() && $variant instanceof \craft\commerce\elements\Variant)
            {
                if($imageAsset = $variant->productImage->one()){
                    $productImages[$lineItem->id] = $imageAsset->getUrl();
                };
            }
        }
        $variables['productImages'] = $productImages;

        $event->variables = $variables;
    }
);
```

Hope that helps.

**Comment by lukeholder** [comment]
Just discovered a bug with the variables not being available to the renderer, so you found a bug. Will fix for the next release. Stay tuned.

**Comment by Stalex89** [comment]
@lukeholder just was about to write that it didn't work for me :)
Thank you and staying tuned!

**Comment by nfourtythree** [comment]
This was fixed in the following commit: 144fffbb7698cf518974b534e454eab048bf0a76

It will be included in the next release of Commerce.

Thanks!

**Comment by Stalex89** [comment]
Thank you for the fix!
